e2e/testsuite/testsuite.go

```diff
@@ -299,13 +299,22 @@ func (s *E2ETestSuite) CreateUserOnChainB(ctx context.Context, amount int64) ibc
 // GetChainANativeBalance gets the balance of a given user on chain A.
 func (s *E2ETestSuite) GetChainANativeBalance(ctx context.Context, user ibc.Wallet) (int64, error) {
 	chainA, _ := s.GetChains()
-	return GetNativeChainBalance(ctx, chainA, user)
+
+	balance, err := s.QueryBalance(ctx, chainA, user.FormattedAddress(), chainA.Config().Denom)
+	if err != nil {
+		return -1, err
```

Comment:
why the special handling here? Is calling `Int64` on a default initialized `Int` failing for some reason? shouldn't it just return `0`?

Reply:
I think returning `0` is fine when `err` is not `nil`

Reply:
We need to handle the error here because `QueryBalance` will return `math.Int{}, err` if err is not nil. It lead to a runtime error if we call `balance.Int64()` here.

Reply:
Thanks the explanation, @trinitys7. But then could we just return 0 instead of -1, as suggested by @DimitrisJim?

Reply:
yes, we can just return 0